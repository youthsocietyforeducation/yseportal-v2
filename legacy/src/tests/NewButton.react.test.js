import React from 'react';
import renderer from 'react-test-renderer';
import { NewButton } from '../NewButton';

test('Link changes the class when hovered', () => {
	const component = renderer.create(<NewButton>Facebook</NewButton>);
	let tree = component.toJSON();
	console.log(tree);
	expect(tree).toMatchSnapshot();

	// // manually trigger the callback
	// tree.props.onMouseEnter();
	// // re-rendering
	// tree = component.toJSON();
	// expect(tree).toMatchSnapshot();

	// // manually trigger the callback
	// tree.props.onMouseLeave();
	// // re-rendering
	// tree = component.toJSON();
	// expect(tree).toMatchSnapshot();
});
