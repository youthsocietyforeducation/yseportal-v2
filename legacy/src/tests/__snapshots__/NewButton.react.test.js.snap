// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="el-tooltip item mr-2"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="el-tooltip__rel"
  >
    <div>
      <button
        className="el-button el-button--success"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          <i
            className="fas fa-plus mr-2"
          />
          New
        </span>
      </button>
    </div>
  </div>
  <div
    className="el-tooltip__popper is-dark"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div>
      
    </div>
    <div
      className="popper__arrow"
    />
  </div>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div
  className="el-tooltip item mr-2"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="el-tooltip__rel"
  >
    <div>
      <button
        className="el-button el-button--success"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          <i
            className="fas fa-plus mr-2"
          />
          New
        </span>
      </button>
    </div>
  </div>
  <div
    className="el-tooltip__popper is-dark"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div>
      
    </div>
    <div
      className="popper__arrow"
    />
  </div>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div
  className="el-tooltip item mr-2"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="el-tooltip__rel"
  >
    <div>
      <button
        className="el-button el-button--success"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          <i
            className="fas fa-plus mr-2"
          />
          New
        </span>
      </button>
    </div>
  </div>
  <div
    className="el-tooltip__popper is-dark"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div>
      
    </div>
    <div
      className="popper__arrow"
    />
  </div>
</div>
`;
